src/libsyntax/ext/expand.rs

```diff
@@ -37,11 +38,15 @@ use std::collections::HashSet;
 
 pub fn expand_expr(e: P<ast::Expr>, fld: &mut MacroExpander) -> P<ast::Expr> {
     let expr_span = e.span;
-    return e.and_then(|ast::Expr {id, node, span}| match node {
+    return e.and_then(|ast::Expr {id, node, span, attrs}| match node {
 
         // expr_mac should really be expr_ext or something; it's the
         // entry-point for all syntax extensions.
         ast::ExprMac(mac) => {
+
+            // FIXME: for now, drop attributes on the macro itself
```

Comment:
hmm okay. I guess I would prefer if the comment made it clear that it is asserting that static statement (that `attrs` is no longer accessed).

I was trying to figure out if there was some important *dynamic* effect you were trying to achieve, e.g. if somehow the memory consumption of the `attrs` had become a concern (perhaps due to cloning?) and thus you were dropping it early as a way to cause the overall memory usage to level out...